sct.c

```diff
@@ -60,42 +59,88 @@ static const struct { float r; float g; float b; } whitepoints[] = {
     { 0.78988728,  0.86491137,  1.00000000, }, /* 10000K */
     { 0.77442176,  0.85453121,  1.00000000, },
 };
-
-static void
-sct_for_screen(Display *dpy, int screen, int temp)
+#define TEMPERATUREDEFAULT 6500
+#define TEMPERATUREMIN     1000
+#define TEMPERATUREMAX     10000
+#define TEMPERATURESTEP    500
+#define GAMMAMULT          65535.0
+#define AVG(c,temp,ratio) whitepoints[(temp) / TEMPERATURESTEP].c * (1 - (ratio)) + whitepoints[(temp) / TEMPERATURESTEP + 1].c * (ratio)
+
+static int get_sct_for_screen(Display *dpy, int screen)
 {
     Window root = RootWindow(dpy, screen);
     XRRScreenResources *res = XRRGetScreenResourcesCurrent(dpy, root);
 
-    temp -= 1000;
-    double ratio = temp % 500 / 500.0;
-#define AVG(c) whitepoints[temp / 500].c * (1 - ratio) + whitepoints[temp / 500 + 1].c * ratio
-    double gammar = AVG(r);
-    double gammag = AVG(g);
-    double gammab = AVG(b);
+    int temp = 0;
+    double t, ts;
+    double gammar, gammag, gammab;
 
-    for (int c = 0; c < res->ncrtc; c++) {
-        int crtcxid = res->crtcs[c];
-        int size = XRRGetCrtcGammaSize(dpy, crtcxid);
+    int n = res->ncrtc;
+    ts = 0.0;
+    if (n > 0)
+    {
+        for (int c = 0; c < n; c++)
+        {
+            int crtcxid = res->crtcs[c];
+            XRRCrtcGamma *crtc_gamma = XRRGetCrtcGamma (dpy, crtcxid);
+            int size = crtc_gamma-> size;
+
+            double g = size / (size - 1) / GAMMAMULT;
```

Comment:
should check that `size >= 2` to prevent a division by `0`

Reply:
Can be made easier:
```
double g = (size + 1) / size / GAMMAMULT;
```

Reply:
```suggestion
            double g_inv = size / (size - 1) / GAMMAMULT;
```
`g_inv` since its the reciprocal of `g` below

Reply:
@zvezdochiot , thanks for your suggestion. In that case, we would add a check `size > 0` to prevent divisions by zero and out-of-bounds array accesses.

Reply:
@faf0 , tell me what to do next. I have never worked with resolved.